R/linelist_merge.R

```diff
@@ -0,0 +1,416 @@
+         names(output_sheet) <- with(var_dict_country,new_var[match(names(output_sheet),old_var_r)])
+      }
+      }
+
+      #handle all date variables
+      require(anytime)
+    #handle dates that are numeric. All character dates -> NA.
+    # Convert to numeric and change date format using origin of excel
+    # Convert to character date for final merge
+    output_sheet$id<-rownames(output_sheet)
+
+    datesnumeric<-output_sheet %>% select(contains("Date", ignore.case = T),id) %>% mutate_all(.funs=gsub,pattern=".",replacement = "/", fixed=T, perl = T) %>%
+      mutate_at(vars(contains("Date", ignore.case = T)), as.numeric) %>%
+      mutate_at(vars(contains("Date", ignore.case = T)),as.Date, origin="1899-12-30") %>%
+      mutate_at(vars(contains("Date", ignore.case = T)), as.character)
+
+      #handle dates that are characters. Select first 10 characters in cell This will select all full dates and not any times appended.
+      # Convert character formats using parse date time.
+      # Convert to numeric and change date format using origin of excel
+      # Convert to character date for final merge
+      datescharacter<-output_sheet %>% select(contains("Date", ignore.case = T), id) %>%
+        mutate_at(vars(contains("Date", ignore.case = T)), lubridate::parse_date_time, orders=c("ymd","Ymd","dmy","dmY","%Y-%m-%d","%y-%m-%d","%d-%m-%y","dBY","ymd HMS","Ymd HMS")) %>%
+        mutate_at(vars(contains("Date", ignore.case = T)),as.Date, origin="1899-12-30") %>% mutate_at(vars(contains("Date", ignore.case = T)),as.character)
+
+
+       #Natural join of data frames, fills in missing values in one df with values from other if not misssing
+
+      dates<- rquery::natural_join(datesnumeric, datescharacter,
+                                   by = "id",
+                                   jointype = "FULL")
+
+      # final date handling: change all columns to date class type (double)
+      dates_final<- dates %>% mutate_at(vars(contains("Date", ignore.case = T)),lubridate::parse_date_time, orders=c("ymd"))
+      # merge in these new handled dates to orginal sheet, removing old coded dates
+      date_vars<-dates_final %>% select(contains("Date", ignore.case = T)) %>% names(.)
+      output_sheet<-output_sheet[ , -which(names(output_sheet) %in% date_vars)]
+      #output_sheet$id<-rownames(output_sheet)
+      output_sheet<- merge(output_sheet,dates_final, by="id") %>% select(-c(id))
+
+
+
+      #remove rows where all columns except ID is missing or if case id is missing
+      output_sheet$na_rows <- rowSums(!is.na(output_sheet))
+      output_sheet<-  output_sheet %>% filter(!is.na(patinfo_id))  %>%  filter(na_rows > 1) %>% select(-c(na_rows))
+
+
+      # make column for ISO code
+      output_sheet$country_iso<-iso
+      #make variable names lower case for output sheet
+      names(output_sheet)<-tolower(names(output_sheet))
+
+  ######
+    #checking that variables of interest exists
```

Comment:
think the way I did [here ](https://github.com/R4IDSR/covidmonitor/pull/9/files#diff-5581348288db36d823a2992aea1e71eb9b0847865fa1e4668c905c5a91734fbdR257) might be shorter?

Reply:
do you mean this for dates or creating missing vars ?

Reply:
made much shorter for creating the missing vars with added warnings etc

Reply:
ah sorry - for some reason links to line references not working. But yes meant for creating missing vars and chucking warnings.